unittests/CppInterOp/ScopeReflectionTest.cpp

```diff
@@ -952,3 +952,29 @@ TEST(ScopeReflectionTest, IncludeVector) {
   )";
   Interp->process(code);
 }
+
+TEST(ScopeReflectionTest, GetBinaryOperator) {
+  Cpp::CreateInterpreter();
+
+  std::string code = R"(
+    class MyClass {
+    public:
+        int x;
+        MyClass(int x) : x(x) {}
+    };
+
+    MyClass operator-(MyClass lhs, MyClass rhs) {
+        return MyClass(lhs.x - rhs.x);
+    }
+
+    MyClass operator+(MyClass lhs, MyClass rhs) {
+        return MyClass(lhs.x + rhs.x);
+    }
+  )";
+
+  Cpp::Declare(code.c_str());
+  EXPECT_TRUE(Cpp::GetBinaryOperator(
```

Comment:
Do we get calls to `GetNamed` with things like `operator+`? If so perhaps we can check if the string starts with `operator` and call `GetBinaryOperator`.

Reply:
> Do we get calls to `GetNamed` with things like `operator+`?

I don't think we get such calls.

---

It may not be that simple. What if the string is `operator_something` or `operator123`? Just checking if it starts with `operator` may not be the correct logic here. Also, `GetBinaryOperator` requires the type string of LHS and RHS.

Reply:
Fair point.